packages/babel-plugin/src/utils/css-builders.ts

```diff
@@ -663,6 +665,44 @@ const extractObjectExpression = (node: t.ObjectExpression, meta: Metadata): CSSO
   return { css: mergeSubsequentUnconditionalCssItems(css), variables };
 };
 
+/**
+ * Extracts CSS data from a member expression node (eg. `styles.primary`)
+ *
+ * @param node Node we're interested in extracting CSS from.
+ * @param meta {Metadata} Useful metadata that can be used during the transformation
+ * @param fallbackToEvaluate {Boolean} Whether to fallback to re-evaluating the expression if it's not a cssMap identifier
+ */
+function extractMemberExpression(
+  node: t.MemberExpression,
+  meta: Metadata,
+  fallbackToEvaluate?: true
+): CSSOutput;
+function extractMemberExpression(
+  node: t.MemberExpression,
+  meta: Metadata,
+  fallbackToEvaluate: false
+): CSSOutput | undefined;
+function extractMemberExpression(
+  node: t.MemberExpression,
+  meta: Metadata,
+  fallbackToEvaluate = true
+): CSSOutput | undefined {
+  const bindingIdentifier = findBindingIdentifier(node);
+  if (bindingIdentifier && meta.state.cssMap[bindingIdentifier.name]) {
+    return {
+      css: [{ type: 'map', expression: node, name: bindingIdentifier.name, css: '' }],
+      variables: [],
+    };
+  }
+
+  if (fallbackToEvaluate) {
```

Comment:
I feel a bit annoyed by adding this `false` here, but I get into this endless loop with non-cssMap objects, eg. `loading ? colors.N20 : colors.N40` where it blew up in a test.

This keeps it 1:1 with the _current state_ by default and lets this `if (bindingIdentifier && meta.state.cssMap[bindingIdentifier.name]) {` flow be reused in `extractConditionalExpression`, so it feels _sane_ to me.